kernel/src/process_standard.rs

```diff
@@ -213,23 +213,35 @@ impl<C: Chip> Process for ProcessStandard<'_, C> {
         self.process_id.get()
     }
 
-    fn enqueue_task(&self, task: Task) -> bool {
+    fn enqueue_task(&self, task: Task) -> Result<(), ErrorCode> {
         // If this app is in a `Fault` state then we shouldn't schedule
         // any work for it.
         if !self.is_active() {
-            return false;
+            return Err(ErrorCode::NODEVICE);
         }
 
-        let ret = self.tasks.map_or(false, |tasks| tasks.enqueue(task));
+        let ret = self.tasks.map_or(Err(ErrorCode::BUSY), |tasks| {
```

Comment:
Should this be `BUSY` or `FAIL`? This is really an internal kernel error that the map cell is empty, no?

Reply:
Yes, but what ErrorCode is that?